packages/form-js-viewer/test/spec/render/components/form-fields/util/valuesUtil.spec.js

```diff
@@ -0,0 +1,180 @@
+      expect(result).to.eql([ { value: 'john', label: 'John' }, { value: 'jessica', label: 'Jessica' } ]);
+    });
+
+
+    it('should add label if not provided', function() {
+
+      // given
+      const values = [ { value: 'john' }, { value: 'jessica' } ];
+
+      // when
+      const result = normalizeValuesData(values);
+
+      // then
+      expect(result).to.eql([ { value: 'john', label: 'john' }, { value: 'jessica', label: 'jessica' } ]);
+    });
+
+
+    it('should ignore valuesData without value', function() {
+
+      // given
+      const valuesData = [ { label: 'John' }, { label: 'Jessica' } ];
+
+      // when
+      const result = normalizeValuesData(valuesData);
+
+      // then
+      expect(result).to.eql([]);
+    });
+
+
+    it('should convert string definitions to value/label objects', function() {
+
+      // given
+      const valuesData = [ 'john', 'jessica' ];
+
+      // when
+      const result = normalizeValuesData(valuesData);
+
+      // then
+      expect(result).to.eql([ { value: 'john', label: 'john' }, { value: 'jessica', label: 'jessica' } ]);
+    });
+
+
+    it('should filter out random objects', function() {
+
+      // given
+      const valuesData = [ { foo: 'bar' }, { value: 'john', label: 'John' } ];
+
+      // when
+      const result = normalizeValuesData(valuesData);
+
+      // then
+      expect(result).to.eql([ { value: 'john', label: 'John' } ]);
+
+    });
+
+
+    it('should allow random objects as values if a proper label is defined', function() {
+
+      // given
+      const valuesData = [ { value: { foo: 'bar', bar: 'foo' }, label: 'myObject' }, { value: 'john', label: 'John' } ];
+
+      // when
+      const result = normalizeValuesData(valuesData);
+
+      // then
+      expect(result).to.eql([ { value: { foo: 'bar', bar: 'foo' }, label: 'myObject' }, { value: 'john', label: 'John' } ]);
+
+    });
+
+
+    it('should filter out random objects as values if no label is defined', function() {
+
+      // given
+      const valuesData = [ { value: { foo: 'bar', bar: 'foo' } }, { value: 'john', label: 'John' } ];
+
+      // when
+      const result = normalizeValuesData(valuesData);
+
+      // then
+      expect(result).to.eql([ { value: 'john', label: 'John' } ]);
+
+    });
+
+
+    it('should convert number definitions to value/label objects', function() {
+
+      // given
+      const valuesData = [ 1, 2 ];
+
+      // when
+      const result = normalizeValuesData(valuesData);
+
+      // then
+      expect(result).to.eql([ { value: 1, label: 1 }, { value: 2, label: 2 } ]);
+    });
+
+
+    it('should handle zero as value', function() {
+
+      // given
+      const valuesData = [ 0 ];
+
+      // when
+      const result = normalizeValuesData(valuesData);
+
+      // then
+      expect(result).to.eql([ { value: 0, label: 0 } ]);
+    });
+
+
+    it('should ignore boolean definitions', function() {
+
+      // given
+      const valuesData = [ true, false ];
```

Comment:
Mh, as I think about this, couldn't that also be a valid use case to specify values for a simple true/false radio? Maybe doesn't make sense, as you could use a checkbox for this, but just in theory.

Reply:
Fair point 🤔, well okay it doesn't hurt much to allow that anyways, booleans have a clear string representation.